### DISCUSSION TITLE
How to detect "reboot"? The event stream doesn't detect it

### DISCUSSION THREAD

**Discussion by alexellis** [cloud-hypervisor/cloud-hypervisor#6273] [Q&A] [answered]
I've used Firecracker extensively where a reboot causes the process to exit, this is very handy for CI workflows and serverless-type workloads.

With cloud-hypervisor, reboot causes the machine to restart like a long-lived server.

There are a few commands that cause an event to be emitted via the event log, but "sudo reboot" or "sudo reboot -f" seem to be ignored.

I have a recollection of discussing this previously, but can't find the issue or discussion.

At the moment, I'm trying to get by with a stream parser that looks for the text: "] reboot: System halted" or "] reboot: Restarting system", then kills the process, but this only works 70-80% of the time, and the way the stream is outputted means sometimes it doesn't work.

**Comment by rbradford** [comment]
@alexellis We consider the event stream as part of our API so we have been conservative about adding events. I think it would be extremely useful to have an event for reboot and shutdown so I would support adding it.

**Reply by rbradford** [reply, marked as answer]
See #6274
